src/sql/azdata.proposed.d.ts

```diff
@@ -1801,5 +1811,69 @@ declare module 'azdata' {
 
 		export interface Dialog extends LoadingComponentBase {
 		}
+
+		/**
+		 * Provides dialog options to customize modal dialog content and layout
+		 */
+		export interface ICustomDialogOptions {
+			/**
+			 * Severity Level to identify icon of modal dialog.
+			 */
+			severity: MessageLevel;
+			/**
+			 * Title of modal dialog header.
+			 */
+			headerTitle: string;
+			/**
+			 * Message text to show on dialog.
+			 */
+			message: string;
+			/**
+			 * (Optional) Detailed message, e.g stack trace of error.
+			 */
+			messageDetails?: string;
+			/**
+			 * (Optional) List of custom actions to include in modal dialog.
+			 */
+			actions?: IDialogAction[];
+			/**
+			 * (Optional) If provided, instruction text is shown in bold below message.
+			 */
+			instructionText?: string;
+			/**
+			 * (Optional) If provided, appends read more link after instruction text.
+			 */
+			readMoreLink?: string;
+		}
+
+		/**
+		 * Provides APIs to define customization for modal dialog action buttons.
+		 */
+		export interface IDialogAction {
```

Comment:
I also wanted to expose a callback method here (similar to IAction.run()), but I wasn't able to receive it in ADS Core. Any suggestions I can try?

Reply:
So you're saying that you want to be able to have the extension register a function that will be called when the specified action is clicked? You'd need to: 

1. Keep a reference of that on the exthost side
2. On the core side when the button is clicked notify the ext host that the dialog had action 'x' clicked
3. The ext host would then run the callback (and return any result as needed to core)

Look at how vscode.InputBoxOptions.validateInput works for an example of similar behavior

Reply:
Thanks! Skipping that for now, will implement later if necessary.